### PR TITLE
Apply transaction for ContractSynchronizer

### PR DESCRIPTION
**Pull request by MixailE**

### CODE REVIEW

**Review by T-Damer — APPROVED** [review]
lgtm

File: src/helpers/EmailBadgeBuilder.ts [hunk 1]
```diff
@@ -14,7 +14,6 @@ export default class EmailBadgeBuilder extends BaseBadgeBuilder {
   }
 
   create(proof: EmailProof) {
-    if (!proof.result) return
```
Comment (smyzin) [inline src/helpers/EmailBadgeBuilder.ts:17]: Why did you remove this check?

File: src/helpers/ERC721BadgeBuilder.ts [hunk 2]
```diff
@@ -14,7 +14,6 @@ export default class ERC721BadgeBuilder extends BaseBadgeBuilder {
   }
 
   create(proof: ERC721Proof) {
-    if (!proof.result) return
```
Comment (smyzin) [inline src/helpers/ERC721BadgeBuilder.ts:17]: Why did you remove this check?

File: src/helpers/ContractSynchronizer.ts [hunk 3]
```diff
@@ -1,15 +1,37 @@
...
+  isTransferEvent,
+  parseLogData,
+} from 'helpers/getOwnedERC721'
 
 export default class ContractSynchronizer {
   account: string
   locked = false
   synchronizedBlockId?: number
   addressToTokenIds: { [address: string]: Set<string> } = {}
+  skipTransactions = new Set<string>()
```
Comment (smyzin) [inline src/helpers/ContractSynchronizer.ts:12]: I don't think we will have a variety of hashes. Do we really need `Set`? If so, okay then

**Review by smyzin — COMMENTED** [review]
Interesting solution

**Review by smyzin — APPROVED** [review]

File: src/helpers/ERC721BadgeBuilder.ts (hunk 2 above)
Comment (MixailE) [reply src/helpers/ERC721BadgeBuilder.ts:17]: I throw an error in mint function

File: src/helpers/ContractSynchronizer.ts (hunk 3 above)
Comment (MixailE) [reply src/helpers/ContractSynchronizer.ts:12]: It would be possible to replace this with an array, I'll think about it
